src/algokit/core/deploy.py

```diff
@@ -0,0 +1,118 @@
+        return genesis_response.json()["network"] if genesis_response.status_code == 200 else None  # noqa: PLR2004
+    except httpx.HTTPError:
+        logger.warning(f"Failed to load network name from {algod_server} due to HTTP error.", exc_info=True)
+        return None
+
+
+def load_deploy_config(network: str, project_dir: Path) -> dict:
+    """
+    Load the deploy configuration for the given network.
+    :param network: Network name.
+    :param project_dir: Project directory path.
+    :return: Deploy configuration.
+    """
+
+    env_file_path = project_dir / f".env.{network}"
+    if not env_file_path.exists():
+        if network in ALGORAND_NETWORKS:
+            logger.info(f"No .env file found at {env_file_path}, loading default configuration for {network}.")
+            return dict(ALGORAND_NETWORKS.get(network, {}))
+
+        raise click.ClickException(f"No .env file found at {env_file_path}")
+
+    # Extract network configuration
+    network_config = dotenv_values(env_file_path, verbose=True, interpolate=True)
+
+    # Fill missing keys with empty strings
+    for required_key in list(AlgorandNetworkConfiguration.__annotations__.keys()):
+        if required_key not in network_config:
+            logger.warning(f"Warning: {required_key} is missing, filling with an empty string.")
+            network_config[required_key] = ""
+
+    return network_config
+
+
+def load_deploy_command(network_name: str, project_dir: Path) -> str:
+    """
+    Load the deploy command for the given network from .algokit.toml file.
+    :param network_name: Network name.
+    :param project_dir: Project directory path.
+    :return: Deploy command.
+    """
+
+    # Load and parse the TOML configuration file
+    config = get_algokit_config(project_dir)
+
+    if not config:
+        raise click.ClickException(
+            f"Couldn't load {ALGOKIT_CONFIG} file. Ensure deploy command is specified, either via "
+            f"--custom-deploy-command or inside {ALGOKIT_CONFIG} file."
+        )
+
+    # Extract the deploy command for the given network
+    try:
+        return str(config["deploy"][network_name]["command"])
+    except KeyError:
+        raise click.ClickException(f"Deploy command is not specified in '{ALGOKIT_CONFIG}' file.") from None
+
+
+def execute_deploy_command(command: str, deploy_config: dict, project_dir: Path | None) -> None:
+    """
+    Execute the deploy command.
+    :param command: Deploy command.
+    :param deploy_config: Deploy configuration.
+    :param project_dir: Project directory path.
+    """
+
+    # Parse the deploy command
+    try:
+        # Get a copy of the current environment variables
+        env = os.environ.copy()
+
+        # Update the environment with custom variables from network_config
+        env.update({key.upper(): str(value) for key, value in deploy_config.items()})
+
+        proc.run(command.split(), env=env, cwd=project_dir or Path.cwd(), stdout_log_level=logging.INFO)
```

Comment:
if you make the suggested changes to `load_deploy_config` method, this method here could become trivial and maybe can be inlined instead, so the env manipulation can be done there

Reply:
This would have been handy if i only needed to load the deploy config and pass it directly to the deploy commands, but if you check the lines 96-113 in deploy.py i also need extra work to be done there to identify if custom network being used for deployment isn't mainnet (if prod flag isn't passed, for extra security considerations it'll prompt for confirmation) additionally i am injecting the mnemonic and optional dispenser. 

Unless you meant something else here i think expecting the deploy config to be passed here instead of implicitly loading within method is still a bit cleaner

LMK what you think @achidlow